src/controller/controller.ts

```diff
@@ -473,6 +473,16 @@ class Controller extends events.EventEmitter {
             const deviceJoinedPayload: Events.DeviceJoinedPayload = {device};
             this.selfAndDeviceEmit(device, Events.Events.deviceJoined, deviceJoinedPayload);
 
+            const deviceInterviewPayload: Events.DeviceInterviewPayload = {status: 'successful', device};
+            this.selfAndDeviceEmit(device, Events.Events.deviceInterview, deviceInterviewPayload);
+        } else if (device.isDeleted) {
```

Comment:
Why was this logic added? How can the device become deleted?

Reply:
When the device is removed from the UI, and then re-paired without restarting Z2M.